File: src/main/java/org/vladg/ast/ASTVisitor.java
package org.vladg.ast;

import org.vladg.ast.repository.model.File;
import org.vladg.ast.service.FileService;
import com.google.inject.Inject;
import lombok.Setter;
import lombok.extern.slf4j.Slf4j;
import org.eclipse.jdt.core.dom.*;

import java.util.List;
import java.util.stream.Collectors;

@Slf4j
public class ASTVisitor extends org.eclipse.jdt.core.dom.ASTVisitor {

    @Setter
    private String currentFileName;

    @Inject
    private FileService fileService;

    @Override
    public boolean visit(TypeDeclaration typeDeclaration) {
        return true;
    }

    public boolean visit(PackageDeclaration packageDeclaration) {
        return true;
    }

    @Override
    public boolean visit(MethodDeclaration methodDeclaration) {
        return true;
    }

    @Override
    public boolean visit(FieldDeclaration fieldDeclaration) {
        return true;
    }

    @Override
    public boolean visit(SingleVariableDeclaration singleVariableDeclaration) {
        return true;
    }

    @Override
    public boolean visit(MethodInvocation node) {
        return true;
    }

    @Override
    public boolean visit(CompilationUnit node) {
        if (node.getModule() != null) {
            return true;
        }
        var imports = ((List<ImportDeclaration>)node.imports())
                                                               .stream()
                                                               .map(it -> it.getName().getFullyQualifiedName())
                                                               .collect(Collectors.toList());
        var astPackage = node.getPackage();
        var packageName = "";
        if (astPackage != null) {
            packageName = astPackage.getName().getFullyQualifiedName();
        }
        this.fileService.addFile(File.builder()
                                     .name(this.currentFileName)
                                     .packageName(packageName)
                                     .imports(imports)
                                     .build());
        return true;
    }

}

File: src/main/java/org/vladg/ast/ASTVisitor.kt
package org.vladg.ast

import com.google.inject.Inject
import org.eclipse.jdt.core.dom.*
import org.eclipse.jdt.core.dom.ASTVisitor
import org.vladg.ast.repository.model.File
import org.vladg.ast.service.FileService
import kotlin.streams.toList

class ASTVisitor : ASTVisitor() {

    @Inject
    private lateinit var fileService: FileService

    override fun visit(typeDeclaration: TypeDeclaration): Boolean {
        return true
    }

    override fun visit(packageDeclaration: PackageDeclaration): Boolean {
        return true
    }

    override fun visit(methodDeclaration: MethodDeclaration): Boolean {
        return true
    }

    override fun visit(fieldDeclaration: FieldDeclaration): Boolean {
        return true
    }

    override fun visit(singleVariableDeclaration: SingleVariableDeclaration): Boolean {
        return true
    }

    override fun visit(node: MethodInvocation): Boolean {
        return true
    }

    override fun visit(node: CompilationUnit): Boolean {
        if (node.module != null) {
            return true
        }
        val imports = (node.imports() as List<ImportDeclaration>)
            .stream()
            .map { it.name.fullyQualifiedName }
            .toList()
        val astPackage: PackageDeclaration? = node.getPackage()
        val packageName = astPackage?.name?.fullyQualifiedName
        fileService.addFile(File(packageName = packageName, imports = imports, name = ""))
        return true
    }
}